lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -798,9 +798,28 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
     std::string Result = Mangler.mangleTypeForDebugger(
         Ty, nullptr);
 
+    // Returns true iff the given type comes from an implicit declaration
+    // generated by the differentiation transform: a linear map struct or
+    // branching trace enum.
+    auto isAutoDiffGeneratedDeclarationType = [](Type type) -> bool {
+      // Peer through metatype types.
+      type = type->getMetatypeInstanceType();
+      // Check for linear map structs.
```

Comment:
Instead of looking at the names of structs, can you instead add a custom attribute?